src/search/search_utils.c

```diff
@@ -0,0 +1,93 @@
+/* Copyright (c) 2013 Jonathan Klee
+
+This file is part of ngp.
+
+ngp is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+ngp is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with ngp.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+
+#include "../utils.h"
+#include "search_utils.h"
+
+
+int validate_file(const char *path)
+{
+    struct stat st;
+
+    if (stat(path, &st) == 0 &&
+        S_ISREG(st.st_mode) == 1)
+        return 1;
+
+    return 0;
+}
+
+const char* apply_regex(const char *output, const char *expr)
+{
+    struct options_t *options = calloc(1, sizeof(*options));
+    char* match = regex(options, output, expr);
+    free( options );
+    return match;
+}
+
+void popen_search(struct search_t *search, external_parser_t *parser)
+{
+    char parser_args[PATH_MAX] = {'\0'};
```

Comment:
You would rather initialize the char array like that `char parser_args[PATH_MAX] = "";`